canonical-web-and-design/vanilla-framework
# Improve class selectors for navigation

## Done

- introduces new navigation class names (`p-navigation__items`, `p-navigation__item`, `p-navigation__link`) to be used in place of existing `p-navigation__links`, `p-navigation__link` and classless `<a>`
- deprecates the use of old classes and selectors
- updates the documentation and examples

Fixes #2168 
Fixes #1746

## QA

- Pull code and `./run`
- Open http://0.0.0.0:8101/ or [demo](https://vanilla-framework-canonical-web-and-design-pr-2852.run.demo.haus/examples)
- Check all navigation examples
- They should work as expected with new classes
- New "deprecated" examples should also work with old classes
- Check [component status](https://vanilla-framework-canonical-web-and-design-pr-2852.run.demo.haus/component-status) - make sure changes to navigation are listed
- Check [navigation documentation](https://vanilla-framework-canonical-web-and-design-pr-2852.run.demo.haus/patterns/navigation) - make sure deprecated classes are documented

Please note - embedded CodePen examples will not work because they use published version of Vanilla, not latest one in development.
